src/frontend/js/utils/AbilitiesHelper/types.ts

```diff
@@ -0,0 +1,13 @@
+import { JoanieUserProfile } from 'types/User';
+
+export enum JoanieUserProfileActions {
+  CAN_ACCESS_TEACHER_DASHBOARD = 'can_access_teacher_dahboard',
+}
+export const isJoanieUserProfileEntity = (entity: Entity): entity is JoanieUserProfile =>
+  JSON.stringify(Object.keys(entity as JoanieUserProfile).sort()) ===
+  JSON.stringify(['abilities', 'full_name', 'id', 'is_staff', 'is_superuser', 'username']);
```

Comment:
I'm not quit sure `sort` method works well if keys contains special chars.
```suggestion
const JOANIE_PROFILE_KEYS = ['abilities', 'full_name', 'id', 'is_staff', 'is_superuser', 'username'];

export const isJoanieUserProfileEntity = (entity: Entity): entity is JoanieUserProfile => {
  const entityKeys = Object.keys(entity);
  
  if (entityKeys.length !== JOANIE_PROFILE_KEYS.length) return false;
  return entityKeys.every((key) => JOANIE_PROFILE_KEYS.includes(key));
};
```